### PR TITLE
DAOS-16837 container: Add client-side DFS metrics

### PR DESCRIPTION
**Pull request by mjmac**
If metrics are enabled for a POSIX container, create
a new pool/$UUID/container/$UUID/dfs metrics root in
the client telemetry to provide DFS-oriented metrics
(POSIX ops, file I/Os, etc).

Also fixes a bug in the agent code for pruning unused
client telemetry segments.

Features: container telemetry
Signed-off-by: Michael MacDonald <mjmac@google.com>

### CODE REVIEW

File: src/client/dfs/obj.c
Comment (mjmac) [inline src/client/dfs/obj.c:419]: @mchaarawi: I will rework this to increment the counters in the relevant `open_` helpers. Not sure what I was thinking here, tbh...

**Comment by mjmac** [comment]
@mchaarawi: I've removed the dependency on the new container property for now, as I'd prefer not to hold this patch up if that becomes a sticking point. Please take a look at let me know if there are any significant changes that you'd like. I can add some ftest coverage for the new DFS metrics if it looks OK.

@ashleypittman: I based this patch on the stats you added to dfuse. It's a bit heavier-weight because it relies on the telemetry library instead of atomics, but maybe there's an opportunity to merge the approaches?

File: src/client/dfs/io.c
Comment (mchaarawi) [inline src/client/dfs/io.c:103]: is this counting the number of read operations?
should you then move it the completion cb too after the op succeeds?
for other operations like mkdir, etc you only increment after the op succeeds not if the op fails.

File: src/client/dfs/io.c
Comment (mchaarawi) [inline src/client/dfs/io.c:167]: so for instance here you increment the nmber of read ops only on success, just FYI.

File: src/client/dfs/obj.c
Comment (mchaarawi) [inline src/client/dfs/obj.c:793]: it sounds like we are incrementing the stat counter twice? once here and once in entry_stat

File: src/client/dfs/obj.c
Comment (mchaarawi) [inline src/client/dfs/obj.c:820]: same here

File: src/client/dfs/obj.c
Comment (mchaarawi) [inline src/client/dfs/obj.c:1027]: i would not make a statx, just stat is fine in this case. statx is just an async version of the API

**Review by mchaarawi — CHANGES_REQUESTED** [review]
just some minor issues that need to be addressed

File: src/client/dfs/obj.c
Comment (mjmac) [reply src/client/dfs/obj.c:1027]: So that comment would probably apply to the other -X calls? I was wondering about that... On the one hand, I can see the simplicity of consolidating them. On the other, I wonder if it's useful to see when a client is using the async versions (performance improvements?). I'll defer to your guidance on that.

Comment (mchaarawi) [reply src/client/dfs/obj.c:1027]: yea i think users care less about the type of the call vs the actual metrics of what is being done. so yes i agree consolidating them with removing the X would be better and would avoid confusion for users between 2 different but really the same metric

File: src/client/dfs/io.c
Comment (mjmac) [reply src/client/dfs/io.c:103]: Oops. I must have screwed up a rebase. I had moved this into the callback. Sigh. I'll go back through and check it all again more carefully.

File: src/client/dfs/obj.c
Comment (mjmac) [reply src/client/dfs/obj.c:793]: Hmm, yep. Removed this one. The increment only happens in entry_stat() if it succeeded.

File: src/client/dfs/obj.c
Comment (mjmac) [reply src/client/dfs/obj.c:820]: Ah, I see what you mean. I'll remove the OSTAT metrics.

File: src/client/dfs/obj.c
Comment (mjmac) [reply src/client/dfs/obj.c]: Done, thanks.

File: src/client/dfs/io.c
Comment (mchaarawi) [inline src/client/dfs/io.c:314]: I would rather you add a condition around this if metrics are enabled only.

**Review by mchaarawi — DISMISSED** [review]
looks good to me.
It would be great if we can enable metrics for something like our existing ior_small.py and mdtest_small.py ci tests and do some verifications there. whether in this PR or other PR is fine with me.

File: src/client/dfs/io.c
Comment (mjmac) [reply src/client/dfs/io.c]: Done.

**Comment by daltonbohning** [comment]
I suspect the copyright GHA is failing because the workflow is coming from a merge of this PR + master, but the source tree used is just this PR.. Something I'll need to consider in the future for new GHA. Anyway, the copyright GHA is not required so can be ignored

**Comment by mjmac** [comment]
> looks good to me. It would be great if we can enable metrics for something like our existing ior_small.py and mdtest_small.py ci tests and do some verifications there. whether in this PR or other PR is fine with me.

I added a simple test to verify that the DFS metrics are incremented as expected after an IOR run. We can add more extensive testing as we get more experience with the feature.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (phender) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:59]: It would be useful to have a description to go with the potential raised exception:
```suggestion
            self.assertGreater(msum, 0, f'{metric} value not greater than zero after I/O')
```
Do we want to verify all metric values before reporting an error?

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (phender) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:64]: Same thing here:
```suggestion
            self.assertGreater(msum, 0, f'{msum} sample_sum not greater than zero after I/O')
```

File: src/tests/ftest/telemetry/dfs_client_telemetry.yaml
Comment (phender) [inline src/tests/ftest/telemetry/dfs_client_telemetry.yaml:30]: FYI, this is now the default.

File: src/tests/ftest/util/telemetry_utils.py
Comment (phender) [inline src/tests/ftest/util/telemetry_utils.py:1266]: Maybe we should handle it this way:
```suggestion
                for metric in metrics[name]['metrics']:
                    if 'labels' not in metric:
                        continue
                    labels = [f'host:{host}']
                    for key, value in metric['labels'].items():
                        labels.append(":".join([str(key), str(value)]))
                    label_key = ",".join(sorted(labels))
                    if 'value' in metric:
                        data[name][label_key] = metric['value']
                    elif 'sample_count' in metric and 'sample_sum' in metric and 'buckets' in metric:
                        data[name][label_key] = {
                            'sample_count': metric['sample_count'],
                            'sample_sum': metric['sample_sum'],
                            'buckets': metric['buckets'],
                        }
                    else:
                        continue
        return data
```
Or will there always be an accompanying `value` entry for each `buckets` entry?

**Review by phender — CHANGES_REQUESTED** [review]
@mjmac @daltonbohning thoughts on failing the test at the first non-expected metric value vs. reporting all non-expected values and then raising a single error? As described in DAOS-15542 (so I could handle this later), the `TelemetryUtils` class has `collect_data()`, `display_data()`, and `verify_data()` methods for reporting and verifying all the telemetry values.  It would probably require a slight modification to the `MetricData` class to support displaying/verifying `sample_sum` instead of `value`.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (daltonbohning) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:58]: FYI python has a built-in for this
```suggestion
            msum = sum(after_metrics[metric].values())
```

File: src/tests/ftest/util/telemetry_utils.py
Comment (daltonbohning) [reply src/tests/ftest/util/telemetry_utils.py:1266]: I disagree with the `else..continue` because that could silently mask errors. I think keeping the implicit assumption is better because it will raise an exception if something is unexpected

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (daltonbohning) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py:59]: Side note: @phender I think we could enforce the description by overriding the function in our base class and making the argument non-optional. Then lint checks would complain about the missing argument - hopefully in the githooks before going through CI

File: src/tests/ftest/telemetry/dfs_client_telemetry.yaml
Comment (mjmac) [reply src/tests/ftest/telemetry/dfs_client_telemetry.yaml:30]: The container type and control_method? Or just the control_method?

File: src/tests/ftest/util/telemetry_utils.py
Comment (mjmac) [reply src/tests/ftest/util/telemetry_utils.py:1266]: > I think keeping the implicit assumption is better because it will raise an exception if something is unexpected

Yeah, that's where I was going with this... The default/standard metric is the `value` type, and I've added a new `histogram` type to allow tests to have access to the histograms that are in the DFS metrics (and will probably be added elsewhere now that they're supported properly).

**Comment by mjmac** [comment]
> > @mjmac @daltonbohning thoughts on failing the test at the first non-expected metric value vs. reporting all non-expected values and then raising a single error? As described in DAOS-15542 (so I could handle this later), the `TelemetryUtils` class has `collect_data()`, `display_data()`, and `verify_data()` methods for reporting and verifying all the telemetry values. It would probably require a slight modification to the `MetricData` class to support displaying/verifying `sample_sum` instead of `value`.
> 
> I think it would be better to handle that separately since this PR is primarily non-ftest. If it is simple to modify the existing function, then I think it's a good idea. But if the function becomes too complex then IMO it's overkill

I'd also prefer to tackle that separately. I think we could probably flesh out the telemetry collection utils a bit more, e.g. create `ValueMetric` and `HistMetric` types with validation, etc. It might also be interesting to add instrumentation to the performance-oriented tests to catch obvious regressions.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (mjmac) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py]: Done.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (daltonbohning) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:29]: Sorry, didn't notice this before. We should add the `dfs` feature tag
```suggestion
        :avocado: tags=dfs,telemetry
```

Comment (mjmac) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py]: Done.

**Comment by mchaarawi** [comment]
unfortunately there was a force-push along the way after my last +1 review and my review diff is not visible anymore. please avoid force-push after someone had reviewed.

**Comment by mjmac** [comment]
> unfortunately there was a force-push along the way after my last +1 review and my review diff is not visible anymore. please avoid force-push after someone had reviewed.

Well, it was still in draft then, so I figured that I'd squash it all down before moving it out of draft and requesting final reviews. In any case, there haven't been any changes to the code outside of ftest since then, other than the check you asked me to add.

**Comment by mchaarawi** [comment]
> > unfortunately there was a force-push along the way after my last +1 review and my review diff is not visible anymore. please avoid force-push after someone had reviewed.
> 
> Well, it was still in draft then, so I figured that I'd squash it all down before moving it out of draft and requesting final reviews. In any case, there haven't been any changes to the code outside of ftest since then, other than the check you asked me to add.

to be fair, you did ask for a review on this PR and i did not jump in to review this myself while it was in draft state ;-)
anyway thanks for the note on the C changes so i don't have to go through them again.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (mchaarawi) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:60]: sounds like the test just verifies that the metrics are > 0 ?
to me that sounds pretty useless test if we are not verifying what is the expected exact value from a deterministic workload.
but again I mentioned that the ftest changes can be done in a future PR. please create a Jira ticket for this so we can tack it.

**Review by phender — CHANGES_REQUESTED** [review]
Need to address the `consider-using-generator` pylint error.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (daltonbohning) [inline src/tests/ftest/telemetry/dfs_client_telemetry.py:59]: I think just don't put brackets, which would make this a generator instead of list
```suggestion
            msum = sum(hist['sample_sum'] for hist in after_metrics[metric].values())
```

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (mjmac) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py:60]: I think that this is better than nothing, but I agree that it would be better to validate that the collected metrics are within expected tolerances. Created https://daosio.atlassian.net/browse/DAOS-16885 to capture this.

File: src/tests/ftest/telemetry/dfs_client_telemetry.py
Comment (daltonbohning) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py:59]: E.g. notice the difference in type here
```
>>> type(i for i in range(3))
<class 'generator'>
>>> type([i for i in range(3)])
<class 'list'>
```

Comment (mjmac) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py]: Ah, I see. Duh. I guess it's more efficient because the consumer (sum()) can do its thing as the values come in vs. waiting for the whole list to be built first and then thrown away.

Comment (daltonbohning) [reply src/tests/ftest/telemetry/dfs_client_telemetry.py]: Right. It doesn't matter much in this case, but if you were using something like `any` where early exit is possible, and a super large "list", it would help.

File: src/tests/ftest/util/telemetry_utils.py
Comment (daltonbohning) [reply src/tests/ftest/util/telemetry_utils.py:1266]: I see what you mean now. Should we handle this as part of the other ftest updates?

File: src/client/dfs/metrics.c
Comment (kjacque) [inline src/client/dfs/metrics.c:32]: Would it make sense to include the pool ID as part of this path?

File: src/tests/ftest/util/telemetry_utils.py
Comment (phender) [inline src/tests/ftest/util/telemetry_utils.py:1265]: This will cause problems with https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/telemetry_utils.py#L1247-L1250 where the `self._data[metric][combined_label]` (i.e. `data[name][label_key]`) value is now a dictionary.

Since the new src/tests/ftest/telemetry/dfs_client_telemetry.py test is not making use of the `self.telemetry.verify_data()` method it wouldn't be an immediate problem, but if future tests verify metrics with sample_count/sample_sum/buckets (or if the metrics for existing tests change to use sample_count/sample_sum/buckets instead of value) will have a problem.

**Review by kjacque — APPROVED** [review]
I didn't look as deeply into the DFS changes. Go and telemetry changes LGTM.

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/telemetry_utils.py
Comment (mjmac) [reply src/tests/ftest/util/telemetry_utils.py:1265]: OK, so what do you suggest as a solution?

File: src/client/dfs/metrics.c
Comment (mjmac) [reply src/client/dfs/metrics.c:32]: > Would it make sense to include the pool ID as part of this path?

... yes. Good catch. I had a concern about whether or not it would work depending on whether or not full client metrics (vs. DFS-only metrics) were enabled, but I tested and it works fine either way. Had to add a new helper to get the pool UUID from the handle.

File: src/tests/ftest/util/telemetry_utils.py
Comment (mjmac) [reply src/tests/ftest/util/telemetry_utils.py:1265]: @daltonbohning, @phender: Would you prefer that I just remove the ftest changes from this PR and we can come up with a more complete solution in DAOS-16885?

**Comment by daltonbohning** [comment]
@mjmac Is this correct? Total ` "sample_count": 2,` but all 4 buckets - 1048575, 2097151, 4194303, +Inf - have count 2?
```
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]             "labels": {
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]               "container": "6981025d-695a-4839-ac13-1e64bd7db15c",
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]               "jobid": "wolf-233vm3.wolf.hpdd.intel.com-84216",
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]               "pid": "84216"
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]             },
2024-12-15 02:49:18,255 process          L0416 DEBUG| [stdout]             "sample_count": 2,
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]             "sample_sum": 1048576,
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]             "buckets": [
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "upper_bound": 255
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,256 process          L0416 DEBUG| [stdout]                 "upper_bound": 511
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "upper_bound": 1023
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]                 "upper_bound": 2047
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,257 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "upper_bound": 4095
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "upper_bound": 8191
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,258 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "upper_bound": 16383
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "upper_bound": 32767
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,259 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "upper_bound": 65535
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "upper_bound": 131071
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,260 process          L0416 DEBUG| [stdout]                 "upper_bound": 262143
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "cumulative_count": 0,
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "upper_bound": 524287
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "cumulative_count": 2,
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]                 "upper_bound": 1048575
2024-12-15 02:49:18,261 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "cumulative_count": 2,
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "upper_bound": 2097151
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "cumulative_count": 2,
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "upper_bound": 4194303
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               },
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               {
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "label": "",
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "cumulative_count": 2,
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]                 "upper_bound": "+Inf"
2024-12-15 02:49:18,262 process          L0416 DEBUG| [stdout]               }
2024-12-15 02:49:18,263 process          L0416 DEBUG| [stdout]             ]
2024-12-15 02:49:18,263 process          L0416 DEBUG| [stdout]           },
```

File: src/tests/ftest/util/telemetry_utils.py
Comment (daltonbohning) [reply src/tests/ftest/util/telemetry_utils.py:1265]: IMO, assuming what you have now is functionally correct, let's keep it for basic code path coverage. Then can expand on it in DAOS-16885

**Comment by daltonbohning** [comment]
> > @mjmac Is this correct? Total ` "sample_count": 2,` but all 4 buckets - 1048575, 2097151, 4194303, +Inf - have count 2?
> 
> Yep. It's a [Cumulative Histogram](https://en.wikipedia.org/wiki/Histogram#Cumulative_histogram). That's [the format](https://www.robustperception.io/why-are-prometheus-histograms-cumulative/) that Prometheus likes.

Interesting :) I don't think I've seen that before

**Comment by mjmac** [comment]
@phender: Nudge... Please advise on how you want me to address your -1 review.

File: src/tests/ftest/util/telemetry_utils.py
Comment (phender) [reply src/tests/ftest/util/telemetry_utils.py:1265]: I have some ideas that could support histogram type metrics as part DAOS-15542.  Since it doesn't seem like it will cause an immediate problem its probably ok.

**Review by phender — APPROVED** [review]
Approving ftest changes.

**Review by kjacque — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**
